test/unified_memory_allocation/memoryPool.hpp

```diff
@@ -73,6 +76,56 @@ TEST_P(umaPoolTest, pow2AlignedAlloc) {
     }
 }
 
+TEST_P(umaPoolTest, allocZeroSize) {
+    static constexpr size_t allocSize = 0;
+    auto *ptr = umaPoolMalloc(pool.get(), allocSize);
+    umaPoolFree(pool.get(), ptr);
+}
+
+TEST_P(umaPoolTest, freeNullptr) {
+    void *ptr = nullptr;
+    umaPoolFree(pool.get(), ptr);
+    ASSERT_EQ(ptr, nullptr);
+}
+
+TEST_P(umaPoolTest, allocOutOfMem) {
+    // test whether memory is kept in a pool accordingly to MaxSize
+    static constexpr size_t allocSize = 16;
+    // MaxSize equals 16 * 1024 * 1024;
+    static constexpr size_t maxAllocSizeMultiplier = 1024 * 1024;
+
+    // allocate until oom
+    auto *ptr = umaPoolMalloc(pool.get(), allocSize);
+    ASSERT_NE(ptr, nullptr);
+
+    for (size_t i = 0; i <= maxAllocSizeMultiplier; ++i) {
+        ptr = umaPoolMalloc(pool.get(), allocSize);
+        ASSERT_NE(ptr, nullptr);
+    }
+
+    umaPoolFree(pool.get(), ptr);
+
+    ptr = umaPoolMalloc(pool.get(), allocSize);
+    ASSERT_NE(ptr, nullptr);
+}
+
+TEST_P(umaPoolTest, multiThreadedMallocFree) {
+    auto poolMalloc = [](size_t allocSize, uma_memory_pool_handle_t pool) {
```

Comment:
It would be better to do multiple allocations in each thread to make sure we test different interleavings - please add some loop here.

Also, there should be some test with different allocation sizes.

Reply:
Done.